.github/actions/slack-action/src/Quote.ts

```diff
@@ -1,7 +1,22 @@
-import { get, HttpError, toHttpTask } from 'ajaxios';
+import { InvalidUrlError, toUrlT, URLParser } from '@execonline-inc/url';
+import { get, Header, HttpError, toHttpTask } from 'ajaxios';
 import Task from 'taskarian';
-import { quoteDecoder } from './Decoders';
-import { ZenQuote } from './Types';
+import { jokeDecoder } from './Decoders';
+import { Joke } from './Types';
 
-export const getZenQuote = (): Task<HttpError, ZenQuote> =>
-  toHttpTask(get('https://zenquotes.io/api/random').withDecoder(quoteDecoder));
+export type QuoteError = InvalidUrlError | HttpError;
+
+const href = 'https://icanhazdadjoke.com/';
+
+const header: Header = { field: 'Accept', value: 'application/json' };
+
+const parseUrlTask = (href: string): Task<QuoteError, URLParser> => toUrlT(href);
+
+export const getJoke = (url: URLParser): Task<QuoteError, Joke> =>
+  toHttpTask(
+    get(url.toString())
+      .withHeader(header)
+      .withDecoder(jokeDecoder),
+  );
+
+export const runJoke = () => parseUrlTask(href).andThen(getJoke);
```

Comment:
I think we could do a bit better with some of the naming
```
export const fetchDadJoke = () => validateUrl(href).andThen(getJoke);
```

Reply:
I agree. Done!